src/algo.rs

```diff
@@ -58,38 +56,47 @@ impl Incorporate for Vec<Profile> {
     fn incorporate(&mut self, candidate: Profile) -> bool {
         // The profiles are ordered in decreasing dep_time
         // The pivot is the element just before the candidate
-        let pivot = self.iter().rposition(|p| p.dep_time >= candidate.dep_time);
-        let mut earlier_profiles: Vec<Profile> = match pivot {
-            None => Vec::new(),
-            Some(position) => self.drain(position + 1..)
-                .filter(|p| candidate.dominates(p))
-                .collect(),
-        };
-
-        let incorporated = candidate.is_non_dominated(self.last());
-        if incorporated {
-            self.push(candidate);
+        match self.iter().rposition(|p| p.dep_time >= candidate.dep_time) {
```

Comment:
https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search may be interesting

Reply:
It would be messierer as I don’t look for a value, but the first element that matches a condition. According to the article, a linear search is faster